### PR TITLE
fix: support additional properties without xDictionaryKey

### PR DESCRIPTION
**Pull request by RicherStAmand**
SamplesMarketplace swagger have this value : 
https://api3-dev.landr.com/samplesmarketplace/swagger/v1/swagger.json#/
```
 "FilterCounters": {
        "type": "object",
        "additionalProperties": false,
        "properties": {
          "sampleTypeFilters": {
            "type": "object",
            "nullable": true,
            "x-dictionaryKey": {
              "$ref": "#/components/schemas/SampleType"
            },
            "additionalProperties": {
              "type": "integer",
              "format": "int64"
            }
          },
          "instrumentFilters": {
            "type": "object",
            "nullable": true,
            "additionalProperties": {
              "type": "integer",
              "format": "int64"
            }
          },
          "genreFilters": {
            "type": "object",
            "nullable": true,
            "additionalProperties": {
              "type": "integer",
              "format": "int64"
            }
          },
          "sfxFilters": {
            "type": "object",
            "nullable": true,
            "additionalProperties": {
              "type": "integer",
              "format": "int64"
            }
          },
          "tagFilters": {
            "type": "object",
            "nullable": true,
            "additionalProperties": {
              "type": "integer",
              "format": "int64"
            }
          }
        }
      },
```

that got type as is : 
```
export interface FilterCounters {
    sampleTypeFilters: {
        [key in SampleType]: number;
    };
}
```

where I was expecting this : 

```
export interface FilterCounters {
    sampleTypeFilters: {
        [key in SampleType]: number;
    },
    instrumentFilters: {[key: string]: number;},
    genreFilters: {[key: string]: number;},
    sfxFilters: {[key: string]: number;},
    tagFilters: {[key: string]: number;}
}
```

I found that we were not properly handling value with `additionalProperties` but without a `xDictionaryKey`.

Part 1 - This PR
Part 2 - https://github.com/LandrAudio/openapi-codegen-typescript/pull/53
Part 3 - https://github.com/LandrAudio/openapi-codegen-typescript/pull/54

### CODE REVIEW

File: src/helpers/getResultStringForAdditionalPropertiesType.ts [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+            res = DataTypes.Number;
+            break;
+        case DataTypes.String:
+            res = DataTypes.String;
+            break;
+        case DataTypes.Array:
+            if (additionalProperties.items && additionalProperties.items[SwaggerProps.$ref]) {
+                res = `${parseRefType(additionalProperties.items[SwaggerProps.$ref].split('/'))}[]`;
+            } else {
+                res = `"// TODO: Something is wrong"`;
```
Comment (SimonDrouin) [inline src/helpers/getResultStringForAdditionalPropertiesType.ts:33]: I wouldnt recommend to put TODOs in code even if its for logging purpose

**Review by SimonDrouin — APPROVED** [review]

File: src/helpers/getResultStringForAdditionalPropertiesType.ts (hunk 1 above)
Comment (RicherStAmand) [reply src/helpers/getResultStringForAdditionalPropertiesType.ts:33]: I agree, but it was already the case and in this PR I focused on adding the support for additional properties type without dictionnary, without changing patterns.

I have upcoming PRs to improve on this.

Comment (RicherStAmand) [reply src/helpers/getResultStringForAdditionalPropertiesType.ts:33]: But at the end of the day, we will still want to write in the result file.
Something more like this ?
`// Error: DataTypes.Array without items or items ref is not supported`

**Review by ardeois — APPROVED** [review]
LGTM
@nemrosim can you check as well, as you are the most familiar with this repo

**Review by pkiler — APPROVED** [review]

**Review by jeremidallaire-landr — APPROVED** [review]

**Review by nemrosim — APPROVED** [review]
LGTM, few comments

File: src/helpers/getResultStringForAdditionalPropertiesType.ts [hunk 2]
```diff
@@ -0,0 +1,42 @@
+import { DataTypes, SwaggerProps } from '../types';
+import { getDescription } from './getDescription';
+import { parseRefType } from './parseRefType';
+
+export const getResultStringForAdditionalPropertiesType = ({
+    additionalProperties,
+    description,
+    propertyName,
+}: {
+    additionalProperties: any;
```
[the remark this answers is not in the document]
Comment (RicherStAmand) [reply src/helpers/getResultStringForAdditionalPropertiesType.ts:10]: `additionalProperties` is returned by `getSchemaProperties`, which type it as `additionalProperties: any`
We don't have any type guard or logic in `typesConverter` to known the types of `additionalProperties` by the time we pass it to `getResultStringForAdditionalPropertiesType`.

[the remark this answers is not in the document]
Comment (RicherStAmand) [reply src/helpers/getResultStringForAdditionalPropertiesType.ts:10]: of course, that could be improve, but in this PR I wanted to have a minimal number of change, just introducing the new helpers needed to resolve the issue I add with the samples marketplace types.

File: tests/getDescription.test.ts [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+describe(`getDescription`, () => {
+    scenarios.forEach(config => {
+        it(`should return ${config.expected} when props is ${config.props}`, async () => {
+            const result = getDescription(config.props);
+
+            expect(result).toBe(config.expected);
+        });
+    });
+});
```
[the remark this answers is not in the document]
Comment (RicherStAmand) [reply tests/getDescription.test.ts]: done

**Pull request merged**